pyhive/hive.py

```diff
@@ -67,32 +67,49 @@ class Connection(object):
     """Wraps a Thrift session"""
 
     def __init__(self, host, port=10000, username=None, database='default', auth='NONE',
-                 configuration=None):
+                 configuration=None, kerberos_service_name=None, password=None):
         """Connect to HiveServer2
 
         :param auth: The value of hive.server2.authentication used by HiveServer2
+        The way to support LDAP and GSSAPI is originated from cloudera/Impyla:
+        https://github.com/cloudera/impyla/blob/255b07ed973d47a3395214ed92d35ec0615ebf62
+        /impala/_thrift_api.py#L152-L160
         """
         socket = thrift.transport.TSocket.TSocket(host, port)
         username = username or getpass.getuser()
         configuration = configuration or {}
-
         if auth == 'NOSASL':
             # NOSASL corresponds to hive.server2.authentication=NOSASL in hive-site.xml
             self._transport = thrift.transport.TTransport.TBufferedTransport(socket)
-        elif auth == 'NONE':
+        elif auth.upper() in ['LDAP', 'KERBEROS', 'NONE']:
+            if auth.upper() == 'KERBEROS' or kerberos_service_name:
+                # KERBEROS mode in hive.server2.authentication is
+                # GSSAPI in sasl library
+                auth = 'GSSAPI'
```

Comment:
I updated other two lines. This line is for Kerberos service that we need. It is a little bit hard to set up hive on kerberos. We are still working on it. I will let you know with any progress we made. Thanks.